packages/flame/lib/src/experimental/geometry/shapes/rectangle.dart

```diff
@@ -123,6 +123,13 @@ class Rectangle extends Shape {
     return Vector2(vx, vy);
   }
 
+  @override
+  Vector2 nearestPoint(Vector2 point) {
+    final xx = (point.x).clamp(_left, _right);
+    final yy = (point.y).clamp(_top, _bottom);
+    return Vector2(xx, yy);
```

Comment:
We have a `Vector2.clamp` method, would it simplify this slightly maybe?

Reply:
not really, because `Vector2.clamp` takes 2 vectors as arguments, so we'd need to create those vectors first...